esp/esdllib/README.md

```diff
@@ -0,0 +1,1608 @@
+    - [BackendRequest](#backendrequest)
+    - [BackendResponse](#backendresponse)
+    - [EsdlResponse](#esdlresponse)
+    - [PreLogging](#prelogging)
+    - [Service](#service)
+  - [XPath](#xpath)
+    - [Custom Functions](#custom-functions)
+      - [ensureDataSection](#ensuredatasection)
+        - [Example](#example)
+      - [getDataSection](#getdatasection)
+        - [Example](#example-1)
+      - [getFeatureSecAccessFlags](#getfeaturesecaccessflags)
+      - [getLogOption](#getlogoption)
+      - [getLogProfile](#getlogprofile)
+      - [getStoredStringValue](#getstoredstringvalue)
+      - [logOptionExists](#logoptionexists)
+      - [secureAccessFlags](#secureaccessflags)
+      - [storedValueExists](#storedvalueexists)
+      - [tokenize](#tokenize)
+      - [Example](#example-2)
+      - [validateFeaturesAccess](#validatefeaturesaccess)
+  - [Operations](#operations)
+    - [add-value](#add-value)
+    - [append-to-value](#append-to-value)
+    - [assert](#assert)
+    - [choose](#choose)
+    - [copy-of](#copy-of)
+    - [ensure-target](#ensure-target)
+    - [fail](#fail)
+    - [for-each](#for-each)
+    - [http-post-xml](#http-post-xml)
+      - [Example](#example-3)
+    - [if](#if)
+    - [if-source](#if-source)
+    - [if-target](#if-target)
+    - [mysql](#mysql)
+    - [namespace](#namespace)
+    - [param](#param)
+    - [remove-node](#remove-node)
+    - [rename-node](#rename-node)
+    - [set-log-option](#set-log-option)
+    - [set-log-profile](#set-log-profile)
+    - [set-value](#set-value)
+    - [source](#source)
+    - [store-value](#store-value)
+    - [target](#target)
+    - [variable](#variable)
+
+<!-- /TOC -->
+
+# Language Overview
+
+ESDL Scripts are written in XML and share some similarites with XSLT, but are easier to understand and have some powerful new features. These are some comparisons and contrasts to XSLT:
+
+1. Scripts use xpaths and support standard, extended and custom xpath functions.
+2. Variables can contain scalar or nodeset values.
+3. You never have to construct a result from scratch, you are _transforming_ an existing element into your desired result. This gives sensible default behavior with no scripting and keeps required scripts smaller.
+4. Does not use the often difficult to trace template matching behavior of xslt; scripts are more easily understood procedural functions.
+
+An ESDL Script is a collection of XML elements and XPath extension functions defined in the `urn:hpcc:esdl:script` namespace, unless otherwise noted. An XML element is either an *entry point* or an *operation*.
+
+An *entry point* is a special container of ordered *operation*. As the ESP processes a transaction, it steps into defined *entry point* in turn and executes its *operations* in order. Each entry point has a *source* XML node that is its default input context, and a *target* XML node that is its default output context. Refer to the [Entry Points](#entry-points) reference for more details.
```

Comment:
Focusing first on "entry point" as a container is a bit confusing to me.  I would probably focus first on the fact the name of an entry point maps to a point in the workflow of an ESDL transaction, and that the element is the root tag of a script.  Then perhaps you can describe it as a container and then refer to the entry point section.

Reply:
Reworded that section